### PR TITLE
Feature kh2 web 1/add footer

### PR DESCRIPTION
**Pull request by ananaple**
[Jira](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=3371&projectKey=KH2WEB&view=detail&selectedIssue=KH2WEB-4)

1) add logo

2) button "join the community"

3) links: icons github, LinkedIn, SoftServe site, about (right side)

image

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,10 +1,14 @@
...
+
 function App() {
   return (
     <div>
-      <h1>
-        What to watch
-      </h1>
+      <Footer />
     </div>
+
```
Comment (dima-diachkov) [inline src/App.tsx:11]: it seems div isn't needed here
```
return (
  <Footer />
)
```
should work correctly

File: src/components/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import "./Footer.css"
+import logo from "./logo.svg";
+import { BottomNavigation, BottomNavigationAction } from "@mui/material";
+import { Container, Grid, Box, Button, Typography } from "@mui/material";
```
Comment (dima-diachkov) [inline src/components/Footer/Footer.tsx:4]: Duplicated import 
Both import from "@mui/material"
I think this should be combined into one

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import LinkedInIcon from '@mui/icons-material/LinkedIn';
+import GitHubIcon from '@mui/icons-material/GitHub';
+import ClosedCaptionOffIcon from '@mui/icons-material/ClosedCaptionOff';
+
+
+
+
+export default function Footer() {
+    return <footer>
+        <Box sx={{position:"fixed", width: "100%", bottom: "0", backgroundColor: "yellowgreen"}}>
```
Comment (dima-diachkov) [inline src/components/Footer/Footer.tsx:14]: Seems like the formatting isn't perfect.

File: src/components/Footer/Footer.css [hunk 4]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (kerrimov) [inline src/components/Footer/Footer.css:1]: remove

File: src/components/Footer/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import "./Footer.css"
+import logo from "./logo.svg";
+import { BottomNavigation, BottomNavigationAction } from "@mui/material";
+import { Container, Grid, Box, Button, Typography } from "@mui/material";
+import LinkedInIcon from '@mui/icons-material/LinkedIn';
+import GitHubIcon from '@mui/icons-material/GitHub';
+import ClosedCaptionOffIcon from '@mui/icons-material/ClosedCaptionOff';
```
Comment (kerrimov) [inline src/components/Footer/Footer.tsx:7]: import in one line

File: src/components/Footer/Footer.tsx (hunk 3 above)
Comment (kerrimov) [reply src/components/Footer/Footer.tsx:14]: move to const

File: src/components/Footer/Footer.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+                        <Box sx={{ width: "10%" }}>
+                            <img src={logo} alt="" />
+                        </Box>
+                        <Box>
+                             <BottomNavigation showLabels>
+                             <Button variant="contained">JOIN THE COMMUNITY</Button>
+                                 <BottomNavigationAction label="LinkedIn" icon={<LinkedInIcon />} />
+                                 <BottomNavigationAction label="GitHub" icon={<GitHubIcon />} />
+                                 <BottomNavigationAction label="SoftServe" icon={<ClosedCaptionOffIcon />} />
+                              </BottomNavigation>
```
Comment (kerrimov) [inline src/components/Footer/Footer.tsx:27]: use Link from mui

File: src/App.tsx (hunk 1 above)
Comment (ananaple) [reply src/App.tsx:11]: fixed

File: src/components/Footer/Footer.tsx (hunk 2 above)
Comment (ananaple) [reply src/components/Footer/Footer.tsx:4]: fixed

File: src/components/Footer/Footer.css (hunk 4 above)
Comment (ananaple) [reply src/components/Footer/Footer.css:1]: done

File: src/components/Footer/Footer.tsx [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+export default function Footer() {
+  return (
+    <footer>
+      <Box sx={style}>
+        <Container maxWidth="lg" sx={{ textAlign: "center" }}>
+          <Grid container spacing={5}>
+            <Grid item xs={12} sm={10}>
+              <Box sx={{ width: "10%" }}>
+                <img src={logo} alt="" />
```
Comment (kerrimov) [inline src/components/Footer/Footer.tsx:23]: where is alt attribute?
why do add box here?

File: src/components/Footer/Footer.tsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+              </Box>
+              <Box>
+              <Typography>
+              At consequatur vero ad odio dolores et ipsum omnis est omnis corporis. Ab fugit eligendi ut voluptatum quae ex asperiores dolore et dolores fugiat. Sed debitis rerum ut aliquid repudiandae sit repellendus quod sed autem porro ut quae facere. Aut quos vitae ut accusamus iusto aut velit assumenda ea laborum internos ut quibusdam quod eos officiis magni et modi.
+              </Typography>
+              </Box>
+              <Box>
+                <BottomNavigation showLabels>
```
Comment (kerrimov) [inline src/components/Footer/Footer.tsx:31]: you don't need bottom navigation, use link

File: src/components/Footer/Footer.tsx [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+                    label="SoftServe"
+                    icon={<ClosedCaptionOffIcon />}
+                  />
+                </BottomNavigation>
+              </Box>
+            </Grid>
+          </Grid>
+          <Box sx={{ textAlign: "center" }}>
+            SoftServe academy &reg; {new Date().getFullYear()}
+          </Box>
```
Comment (kerrimov) [inline src/components/Footer/Footer.tsx:53]: don't need box here

File: src/components/Footer/Footer.tsx [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+            <Box>
+              <Typography component="p" paragraph>
+                WTW invites you to choose a movie for the evening or make a wish
+                list to watch
+              </Typography>
+              <Typography component="p" paragraph>
+                This entertainment site was created for educational purposes.
+                The <Link href="https://developers.themoviedb.org/">API</Link>{" "}
+                used is freely available.
+              </Typography>
```
Comment (kerrimov) [inline src/components/Footer/Footer.tsx:87]: use 1 typography

File: src/components/Footer/Footer.tsx [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+export default function Footer() {
+  return (
+    <Box sx={style.root}>
+      <Container maxWidth="lg">
+        <Grid container spacing={2}>
+          <Grid item xs>
+            <Box>
+              <Avatar variant="square" src={logo} sx={style.avatar} />
+            </Box>
```
Comment (kerrimov) [inline src/components/Footer/Footer.tsx:56]: why do you need box here?

File: src/components/Footer/Footer.tsx [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+              <Button color="inherit" href="https://uk-ua.facebook.com/">
+                <Facebook fontSize="large" />
+              </Button>
+              <Button color="inherit" href="https://www.linkedin.com/">
+                <LinkedIn fontSize="large" />
+              </Button>
+              <Button color="inherit" href="https://github.com/kerrimov/what-to-watch">
+                <GitHub fontSize="large" />
+              </Button>
+            </Box>
```
Comment (kerrimov) [inline src/components/Footer/Footer.tsx:75]: use map and move to separete component

File: src/components/Footer/Footer.tsx [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+function Copyright() {
+  return (
+    <Typography variant="body2" color="text.secondary">
+      {"Copyright © "}
+      <Link color="inherit" href="https://softserveinc.com/">
+        SoftServe
+      </Link>{" "}
+      {new Date().getFullYear()}
+      {"."}
```
Comment (kerrimov) [inline src/components/Footer/Footer.tsx:22]: what is `{" "}` `{"."}`

File: src/components/Footer/Footer.tsx [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+  Typography,
+  Link,
+  Avatar,
+} from "@mui/material";
+import { Facebook, GitHub, LinkedIn } from "@mui/icons-material";
+
+function Copyright() {
+  return (
+    <Typography variant="body2" color="text.secondary">
+      {"Copyright © "}
```
Comment (kerrimov) [inline src/components/Footer/Footer.tsx:17]: why do you need curly braces?

File: src/components/Footer/Footer.tsx [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+  Grid,
+  Box,
+  Button,
+  Typography,
+  Link,
+  Avatar,
+} from "@mui/material";
+import { Facebook, GitHub, LinkedIn } from "@mui/icons-material";
+
+function Copyright() {
```
Comment (kerrimov) [inline src/components/Footer/Footer.tsx:14]: move to separete file

File: src/components/Footer/Footer.tsx (hunk 13 above)
Comment (ananaple) [reply src/components/Footer/Footer.tsx:22]: {" "} - is for whitespace between SoftServe and 2022
{"."} - removed

File: src/components/Footer/Footer.tsx (hunk 14 above)
Comment (ananaple) [reply src/components/Footer/Footer.tsx:17]: its for whitespace, removed curly braces and added {" "} instead

File: src/components/Footer/Footer.tsx [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+            justifyContent="center"
+            marginRight={10}
+          >
+            <Social />
+          </Grid>
+          <Grid item lg>
+            <Box>
+              <Typography component="p" paragraph gutterBottom>
+                WTW invites you to choose a movie for the evening or make a wish
+                list to watch <br /><br />
```
Comment (kerrimov) [inline src/components/Footer/Footer.tsx:49]: `<br /><br />` - Looks messy

File: src/components/Footer/Footer.tsx (hunk 13 above)
Comment (ananaple) [reply src/components/Footer/Footer.tsx]: fixed, please check

File: src/components/Footer/Footer.tsx [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+    bottom: "0",
+  },
+  avatar: {
+    width: "150px",
+    height: "150px",
+  },
+};
+
+export default function Footer() {
+  return (
```
[the remark this answers is not in the document]
Comment (ananaple) [reply src/components/Footer/Footer.tsx]: fixed

File: src/components/Footer/Footer.tsx [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+  Container,
+  Grid,
+  Box,
+  Typography,
+  Link,
+  Avatar,
+} from "@mui/material";
+
+const style = {
+  root: {
```
[the remark this answers is not in the document]
Comment (ananaple) [reply src/components/Footer/Footer.tsx]: done

File: src/components/Footer/Footer.tsx [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+            item
+            xs={4}
+            alignItems="center"
+            justifyContent="center"
+            marginRight={10}
+          >
+            <Social />
+          </Grid>
+          <Grid item lg>
+            <Box>
```
[the remark this answers is not in the document]
Comment (ananaple) [reply src/components/Footer/Footer.tsx]: no need, removed

File: src/components/Footer/Footer.tsx (hunk 16 above)
Comment (ananaple) [reply src/components/Footer/Footer.tsx]: removed

File: src/App.tsx [hunk 20]
```diff
@@ -1,11 +1,15 @@
...
 
 function App() {
   return (
-    <h1>
-      What to watch
-    </h1>
+    <ThemeProvider theme={customTheme}>
+      <Footer />
+    </ThemeProvider>
```
Comment (kerrimov) [inline src/App.tsx:11]: These changes are in the index.tsx.

File: src/components/Footer/components/Copyright.tsx [hunk 21]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import { Typography, Link } from "@mui/material";
+
+export default function Copyright() {
+  
+
```
Comment (kerrimov) [inline src/components/Footer/components/Copyright.tsx:6]: remove unnecessary space

File: src/components/Footer/components/Copyright.tsx [hunk 22]
```diff
@@ -0,0 +1,16 @@
...
+  return (
+    <Typography variant="body2" color="text.secondary">
+     {`Copyright @ `}
```
Comment (kerrimov) [inline src/components/Footer/components/Copyright.tsx:9]: it is not interpolation

File: src/components/Footer/components/Social.tsx [hunk 23]
```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import { Button, Box } from "@mui/material";
+import { socialLinks } from "../mocks/LinksData";
+
```
Comment (kerrimov) [inline src/components/Footer/components/Social.tsx:4]: 1 space is enough

File: src/components/Footer/components/Social.tsx [hunk 24]
```diff
@@ -0,0 +1,31 @@
...
+        <IconName color="success" fontSize="large" />
+      </Button>
+      ))}
+    </Box>
+  );
+};
+
+export default Social;
+
+
```
Comment (kerrimov) [inline src/components/Footer/components/Social.tsx:31]: remove

File: src/App.tsx (hunk 20 above)
Comment (ananaple) [reply src/App.tsx:11]: removed

File: src/components/Footer/components/Copyright.tsx (hunk 22 above)
Comment (ananaple) [reply src/components/Footer/components/Copyright.tsx:9]: i made interpolation in variable  fullDate
but i need this {` `} for whitespaces inside Typography

File: src/components/Footer/components/Social.tsx (hunk 23 above)
Comment (ananaple) [reply src/components/Footer/components/Social.tsx:4]: fixed

File: src/components/Footer/components/Social.tsx (hunk 24 above)
Comment (ananaple) [reply src/components/Footer/components/Social.tsx:31]: done

File: src/components/Footer/components/Copyright.tsx [hunk 25]
```diff
@@ -0,0 +1,20 @@
...
+
+  return (
+    <Box>
+      <Typography variant="body2" color="text.secondary">
+          Copyright @
+        <Link color="inherit" underline="hover" href="https://softserveinc.com/">
+          {` SoftServe `}
+        </Link> 
+          {fullDate}
+      </Typography>
```
Comment (kerrimov) [inline src/components/Footer/components/Copyright.tsx:17]: use flex to escape adding space like this

Comment (ananaple) [reply src/components/Footer/components/Copyright.tsx]: done. Please check

File: src/components/Footer/components/Copyright.tsx [hunk 26]
```diff
@@ -0,0 +1,20 @@
...
+import { Box, Typography, Link } from "@mui/material";
+
+export default function Copyright() {
+  
+  const date = new Date();
+  const fullDate = `${date.getFullYear()}`;
+
+  return (
+    <Box>
+      <Typography variant="body2" color="text.secondary" display="flex" width="190px" sx={{justifyContent: "space-evenly"}}>
```
Comment (kerrimov) [inline src/components/Footer/components/Copyright.tsx:11]: move to const styles

File: src/components/Footer/components/Copyright.tsx [hunk 27]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { Box, Typography, Link } from "@mui/material";
+
+export default function Copyright() {
```
Comment (kerrimov) [inline src/components/Footer/components/Copyright.tsx:4]:
```suggestion
export const  Copyright = () => {
```

**Review by kerrimov — COMMENTED** [review]
add index.ts file to every component

File: src/components/Footer/components/Copyright.tsx (hunk 26 above)
Comment (ananaple) [reply src/components/Footer/components/Copyright.tsx]: done

File: src/components/Footer/components/Copyright.tsx (hunk 27 above)
Comment (ananaple) [reply src/components/Footer/components/Copyright.tsx]: done

File: src/components/Footer/Footer.tsx [hunk 28]
```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import logo from "./logo.svg";
+import Social from "./Social/Social";
+import { Copyright } from "./Copyright/Copyright";
```
Comment (kerrimov) [inline src/components/Footer/Footer.tsx:4]: import from the folder root as you have index file

**Review by kerrimov — COMMENTED** [review]
resolve conflicts with master

File: src/components/Footer/Footer.tsx (hunk 28 above)
Comment (ananaple) [reply src/components/Footer/Footer.tsx]: fixed

File: yarn.lock [hunk 29]
```diff
@@ -1578,10 +1578,17 @@
...
 "@mui/icons-material@^5.8.3":
   version "5.8.3"
   resolved "https://registry.yarnpkg.com/@mui/icons-material/-/icons-material-5.8.3.tgz#75c8bde42e6ba71e3871439a2b751be987343493"
   integrity sha512-dAdhimSLKOV0Q8FR7AYGEaCrTUh9OV7zU4Ueo5REoUt4cC3Vy+UBKDjZk66x5ezaYb63AFgQIFwtnZj3B/QDbQ==
+=======
+"@mui/icons-material@^5.8.2":
+  version "5.8.2"
+  resolved "https://registry.yarnpkg.com/@mui/icons-material/-/icons-material-5.8.2.tgz#211ac2a041ece952749d39c3c26071226993be14"
+  integrity sha512-fP6KUCCZZjc2rdbMSmkNmBHDskLkmP0uCox57cbVXvomU6BOPrCxnr5YXsSsQrZB8fchx7hfH0bkAgvMZ5KM0Q==
+>>>>>>> 4ed0eac (first commit)
```
Comment (kerrimov) [inline yarn.lock:1591]: resolve conflicts

**Review by kerrimov — COMMENTED** [review]
squash commits

File: yarn.lock (hunk 29 above)
Comment (ananaple) [reply yarn.lock]: done, please check

**Review by kerrimov — APPROVED** [review]

**Comment by ananaple** [comment]
megring and rebasing

**Pull request merged**